test/albumController.test.js

```diff
@@ -173,24 +155,151 @@ describe('AlbumController API Tests', () => {
 
     await Album.insertMany(albumsToCreate);
 
-    // Make a GET request to the /api/v1/albums/filter endpoint with a limit parameter set to 4
     const limit = 3;
     const response = await request(app).get(
       `/api/v1/albums/filter?limit=${limit}`
     );
 
-    // Assert that the response status is OK (200)
     expect(response.status).toBe(StatusCodes.OK);
-
-    // Assert that the response body is an object with "albums" and "count" keys
     expect(response.body).toHaveProperty('albums');
     expect(response.body).toHaveProperty('count');
-
-    // Assert that "albums" is an array with a length equal to the specified limit
     expect(response.body.albums).toHaveLength(limit);
+    expect(response.body.count).toBe(albumsToCreate.length);
+  });
 
-    // Assert that the "count" field reflects the actual count of albums in the database
+  it('should create a new album - Success Case', async () => {
+    const adminCredentials = {
+      email: 'admin@admin.com',
+      password: 'adminpassword',
+    };
 
-    expect(response.body.count).toBe(albumsToCreate.length);
+    const signedCookie = await loginAndReturnCookie(adminCredentials);
+
+    const newAlbumData = {
+      albumName: 'New Album',
+      artistName: 'New Artist',
+      price: 9.99,
+      spotifyUrl: 'https://api.spotify.com/v1/albums/new123',
+    };
+
+    const response = await request(app)
+      .post('/api/v1/albums')
+      .set('Cookie', signedCookie)
+      .send(newAlbumData);
+
+    expect(response.status).toBe(StatusCodes.CREATED);
+    expect(response.body).toHaveProperty('album');
```

Comment:
It would be good to also have an assertion to verify that the returned album is what we expect. You could test that `response.body.album.albumName` matches what we put into the request.

Or you could use the `toMatchObject` [Jest matcher](https://jestjs.io/docs/expect#tomatchobjectobject), to verify that `newAlbumData` is a subset of the returned album in the response body: `expect(response.body.album).toMatchObject(newAlbumData);`

Reply:
Akos, I have added verification that the returned album matches the expected data using .toMatchObject(object), commit ae6985f501b4a99e161bed85decd056994ea1d52